website/docs/guides/global-events.md

```diff
@@ -0,0 +1,115 @@
+---
+sidebar_position: 6
+---
+
+# Global Events
+
+In this section, we will delve into the concept of Global Events in KafkaFlow, which provides a mechanism to subscribe to various events that are triggered during the message production and consumption processes. 
+
+KafkaFlow offers a range of Global Events that can be subscribed to. These events can be used to monitor and react to different stages of message handling. Below is a list of available events:
+  - [Message Produce Started Event](#message-produce-started-event)
+  - [Message Produce Completed Event](#message-produce-completed-event)
+  - [Message Produce Error Event](#message-produce-error-event)
+  - [Message Consume Started Event](#message-consume-started-event)
+  - [Message Consume Completed Event](#message-consume-completed-event)
+  - [Message Consume Error Event](#message-consume-error-event)
+
+## Message Produce Started Event {#message-produce-started-event}
+
+The Message Produce Started Event is triggered when the message production process begins. It provides an opportunity to perform tasks or gather information before the message is sent.
+
+```csharp
+services.AddKafka(
+    kafka => kafka
+        .SubscribeGlobalEvents(observers =>
+        {
+            observers.MessageProduceStarted.Subscribe(eventContext =>
+            {
+                // Add your logic here
+            });
+        })
+```
+
+## Message Produce Completed Event {#message-produce-completed-event}
+
+The Message Produce Completed Event occurs when a message is successfully produced. Subscribing to this event enables you to track the successful completion of message production.
+
+```csharp
+services.AddKafka(
+    kafka => kafka
+        .SubscribeGlobalEvents(observers =>
+        {
+            observers.MessageProduceCompleted.Subscribe(eventContext =>
+            {
+                // Add your logic here
+            });
+        })
+```
+
+## Message Produce Error Event {#message-produce-error-event}
+
+In case an error occurs during message production, the Message Produce Error Event is triggered. Subscribing to this event allows you to handle errors gracefully and perform, for instance, any necessary cleanup or logging.
```

Comment:
"By subscribing to this event, you will be able to catch any exceptions that may occur while producing a message."

Reply:
Done, please check.